compiler/rustc_mir_build/src/build/expr/into.rs

```diff
@@ -268,6 +268,8 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                         // FIXME(varkor): replace this with an uninhabitedness-based check.
                         // This requires getting access to the current module to call
                         // `tcx.is_ty_uninhabited_from`, which is currently tricky to do.
+                        // FIXME(Amanieu): Make sure to also update FnAbi::noreturn when
```

Comment:
We don't have to wait - `FnAbi` can rely on details that we don't expose to borrowck (the reason MIR needs this weird limitation is because borrowck could tell there is dead code after the call, I believe - basically a back-compat hazard if the type is *privately* uninhabited, but there is no compat limitation on codegen).